// To compile this file, run this inside the file's directory:
//   protoc --python_out=./protobuf newaddressbook.proto

/*
    Python proto buffer compiler doesn't generate data access code directly.
*/

syntax = "proto2";

package tutorial;

message NewPerson {
    optional string name = 1;
    optional int32 id = 2;
    optional string email = 3;

    enum NewPhoneType {
        PHONE_TYPE_UNSPECIFIED = 0;
        PHONE_TYPE_MOBILE = 1;
        PHONE_TYPE_HOME = 2;
        PHONE_TYPE_WORK = 3;
    }

    message NewPhoneNumber {
        optional string number = 1;
        optional NewPhoneType type = 2 [default = PHONE_TYPE_HOME];
    }

    repeated NewPhoneNumber phones = 4;
}

message NewAddressBook {
    optional int32 id = 1;
    repeated NewPerson people = 2;
}
